src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonLexer.cs

```diff
@@ -0,0 +1,413 @@
+        {
+            Text = text;
+        }
+
+        public VirtualChar CurrentChar => Position < Text.Length ? Text[Position] : new VirtualChar((char)0, default);
+
+        public ImmutableArray<VirtualChar> GetSubPattern(int start, int end)
+        {
+            var result = ArrayBuilder<VirtualChar>.GetInstance(end - start);
+            for (var i = start; i < end; i++)
+            {
+                result.Add(Text[i]);
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        public JsonToken ScanNextToken()
+        {
+            var leadingTrivia = ScanTrivia(leading: true);
+            if (Position == Text.Length)
+            {
+                return CreateToken(
+                    JsonKind.EndOfFile, leadingTrivia, 
+                    ImmutableArray<VirtualChar>.Empty, ImmutableArray<JsonTrivia>.Empty);
+            }
+
+            var (chars, kind, diagnostic) = ScanNextTokenWorker();
+            Debug.Assert(chars.Length > 0);
+
+            var trailingTrivia = ScanTrivia(leading: false);
+            var token = CreateToken(kind, leadingTrivia, chars, trailingTrivia);
+
+            if (diagnostic != null)
+            {
+                token = token.AddDiagnosticIfNone(diagnostic.Value);
+            }
+
+            return token;
+        }
+
+        private bool IsSpecial(char ch)
+        {
+            // Standard tokens.
+            switch (ch)
+            {
+                case '{': case '}':
+                case '[': case ']':
+                case '(': case ')':
+                case ',': case ':': 
+                case '\'': case '"':
+                    return true;
+
+                case ' ': case '\t': case '/': case '\r': case '\n':
+                    // trivia cases
+                    return true;
+            }
+
+            // more trivia
+            if (char.IsWhiteSpace(ch))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private (ImmutableArray<VirtualChar>, JsonKind, EmbeddedDiagnostic? diagnostic) ScanNextTokenWorker()
+        {
+            Debug.Assert(Position < Text.Length);
+            switch (this.CurrentChar)
+            {
+                case '{': return ScanSingleCharToken(JsonKind.OpenBraceToken);
+                case '}': return ScanSingleCharToken(JsonKind.CloseBraceToken);
+                case '[': return ScanSingleCharToken(JsonKind.OpenBracketToken);
+                case ']': return ScanSingleCharToken(JsonKind.CloseBracketToken);
+                case '(': return ScanSingleCharToken(JsonKind.OpenParenToken);
+                case ')': return ScanSingleCharToken(JsonKind.CloseParenToken);
+                case ',': return ScanSingleCharToken(JsonKind.CommaToken);
+                case ':': return ScanSingleCharToken(JsonKind.ColonToken);
+
+                case '\'': case '"':
+                    return ScanString();
+
+                //case '-': case '.':
```

Comment:
Why is this commented out?

Reply:
Added comment explaining why we do not explicitly lex out numbers here, but instead let the parser take control.  #resolved.